plugins/web-worker-offloading/hooks.php

```diff
@@ -0,0 +1,133 @@
+<?php
+/**
+ * Hook callback for Web Worker Offloading.
+ *
+ * @since n.e.x.t
+ * @package web-worker-offloading
+ */
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit; // Exit if accessed directly.
+}
+
+/**
+ * Gets configuration for Web Worker Offloading.
+ *
+ * @since n.e.x.t
+ * @link https://partytown.builder.io/configuration
+ *
+ * @return array{ debug?: bool, forward?: non-empty-string[], lib: non-empty-string, loadScriptsOnMainThread?: non-empty-string[], nonce?: non-empty-string } Configuration for Partytown.
+ */
+function wwo_get_configuration(): array {
+	$config = array(
+		'lib'     => wp_parse_url( plugin_dir_url( __FILE__ ), PHP_URL_PATH ) . 'build/',
+		'forward' => array(),
+	);
+
+	/**
+	 * Add configuration for Web Worker Offloading.
+	 *
+	 * @since n.e.x.t
+	 * @link https://partytown.builder.io/configuration
+	 *
+	 * @param array{ debug?: bool, forward?: non-empty-string[], lib: non-empty-string, loadScriptsOnMainThread?: non-empty-string[], nonce?: non-empty-string } $config Configuration for Partytown.
+	 */
+	return apply_filters( 'wwo_configuration', $config );
+}
+
+/**
+ * Initialize Web Worker Offloading.
+ *
+ * @since n.e.x.t
+ */
+function wwo_init(): void {
+	$partytown_js = file_get_contents( __DIR__ . '/build/partytown.js' ); // phpcs:ignore WordPress.WP.AlternativeFunctions.file_get_contents_file_get_contents -- It's a local filesystem path not a remote request.
+	if ( false === $partytown_js ) {
+		return;
+	}
+
+	wp_register_script(
+		'web-worker-offloading',
+		'',
+		array(),
+		WEB_WORKER_OFFLOADING_VERSION,
+		array( 'in_footer' => false )
+	);
+
+	wp_add_inline_script(
+		'web-worker-offloading',
+		sprintf(
+			'window.partytown = %s;',
+			wp_json_encode( wwo_get_configuration() )
+		),
+		'before'
+	);
+
+	wp_add_inline_script( 'web-worker-offloading', $partytown_js );
+}
+add_action( 'wp_enqueue_scripts', 'wwo_init' );
+
+/**
+ * Mark scripts with `web-worker-offloading` dependency as async.
+ *
+ * Why this is needed?
+ *
+ * Scripts offloaded to a worker thread can be considered async. However, they may include `before` and `after` inline
+ * scripts that need sequential execution. Once marked as async, `filter_eligible_strategies()` determines if the
+ * script is eligible for async execution. If so, it will be offloaded to the worker thread.
+ *
+ * @since n.e.x.t
+ *
+ * @param string[] $script_handles Array of script handles.
+ * @return string[] Array of script handles.
+ */
+function wwo_update_script_strategy( array $script_handles ): array {
+	foreach ( $script_handles as $handle ) {
+		if ( in_array( 'web-worker-offloading', wp_scripts()->registered[ $handle ]->deps, true ) ) {
```

Comment:
Some hardening in case (1) another filter fails to return an array, and (2) a filter returns a handle for a script that was unregistered (or was never registered):

```suggestion
 * @param string[]|mixed $script_handles Array of script handles.
 * @return string[] Array of script handles.
 */
function wwo_update_script_strategy( $script_handles ): array {
	$script_handles = array_intersect( (array) $script_handles, array_keys( wp_scripts()->registered ) );
	foreach ( $script_handles as $handle ) {
		if ( in_array( 'web-worker-offloading', wp_scripts()->registered[ $handle ]->deps, true ) ) {
```